sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/DataSourceSuite.scala

```diff
@@ -109,18 +109,22 @@ class DataSourceSuite extends SharedSparkSession with PrivateMethodTester {
   }
 
   test("test non existent paths") {
-    assertThrows[AnalysisException](
-      DataSource.checkAndGlobPathIfNecessary(
-        Seq(
-          path1.toString,
-          path2.toString,
-          nonExistentPath.toString
-        ),
-        hadoopConf,
-        checkEmptyGlobPath = true,
-        checkFilesExist = true,
-        enableGlobbing = true
-      )
+    checkError(
+      exception = intercept[AnalysisException](
+        DataSource.checkAndGlobPathIfNecessary(
+          Seq(
+            path1.toString,
+            path2.toString,
+            nonExistentPath.toString
+          ),
+          hadoopConf,
+          checkEmptyGlobPath = true,
+          checkFilesExist = true,
+          enableGlobbing = true
+        )
+      ),
+      errorClass = "PATH_NOT_FOUND",
+      parameters = Map("path" -> "mockFs://mockFs/nonexistentpath")
```

Comment:
@itholic can you use `nonExistentPath` variable here?

Reply:
seems to be related to the order of path check, the `parameters` is filePath1, but the actual is `filePath3` with Scala 2.13

Reply:
Sure, let me create a follow-up.

Thanks!

Reply:
https://github.com/apache/spark/blob/f692771444d72bbac80abf6ac6066059b0ac4579/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala#L770-L781

@itholic It seems that the check is parallel, and the results may be different for different Scala version. If we don't care about this, can we consider making the case only have `nonExistentPath `?